### PR TITLE
feat (components): add catalogue components to tailwind component lib

### PR DESCRIPTION
**Pull request by MaxPostema**
What are the main changes you did:
- extracted tailwind components from #4031

how to test:
- all the new components work
 
todo:
- [ ] updated docs in case of new feature
- [ ] added/updated tests
- [ ] added/updated testplan to include a test for this fix, including ref to bug using # notation

### CODE REVIEW

File: apps/tailwind-components/components/content/type/ContentTypeRefBack.vue [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      <TableHeadRow>
+        <TableHead v-for="longestRowColumns in rows[longestRowIndex]">{{
+          longestRowColumns.key
+        }}</TableHead>
+      </TableHeadRow>
+    </template>
+    <template #body>
+      <TableRow v-for="row in rows">
+        <TableCell v-for="longestRowColumns in rows[longestRowIndex]">{{
+          row.find((col: any) => col.key === longestRowColumns.key)?.value
```
Comment (davidruvolo51) [inline apps/tailwind-components/components/content/type/ContentTypeRefBack.vue:13]: is the type for `col` a `Record`?

File: apps/tailwind-components/components/content/type/ContentTypeString.vue [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<template>
+  <p>{{ field.value }}</p>
+</template>
+
+<script setup lang="ts">
+import type { ISectionField } from "~/interfaces/types";
```
Comment (davidruvolo51) [inline apps/tailwind-components/components/content/type/ContentTypeString.vue:6]: Is this file available? To organise the files, should we use interfaces or types?

File: apps/tailwind-components/components/content/type/ContentTypeText.vue [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<template>
+  <ContentReadMore :text="field.value" :cutoff="500" />
+</template>
+
+<script setup lang="ts">
+import type { ISectionField } from "~/interfaces/types";
```
Comment (davidruvolo51) [inline apps/tailwind-components/components/content/type/ContentTypeText.vue:6]: Same as the previous comments

File: apps/tailwind-components/components/display/List.vue [hunk 4]
```diff
@@ -1,3 +1,15 @@
+<template>
+  <section>
+    <h3 v-if="title" class="mb-2.5 font-bold text-body-base">{{ title }}</h3>
+    <ul
+      class="columns-1 text-body-base"
+      :class="`${columnCountClass} ${listClasses} `"
```
Comment (davidruvolo51) [inline apps/tailwind-components/components/display/List.vue:6]: The JS code should be moved in into the class attribute

File: apps/tailwind-components/components/display/ListItem.vue [hunk 5]
```diff
@@ -1,3 +1,7 @@
+<template>
+  <li :class="listItemClasses"><slot></slot></li>
```
Comment (davidruvolo51) [inline apps/tailwind-components/components/display/ListItem.vue:2]: If possible, the css classes should be defined inline

File: apps/tailwind-components/pages/DataFetch.other.vue [hunk 6]
```diff
@@ -0,0 +1,123 @@
+<template>
+  <h1>Data and Meta data</h1>
+  <div class="h-12 mt-4 mb-16">
+    <h3 class="text-heading-lg">Params</h3>
```
Comment (davidruvolo51) [inline apps/tailwind-components/pages/DataFetch.other.vue:4]: This should be a `h2` element

File: apps/tailwind-components/pages/DataFetch.other.vue [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+    <div class="m-2">
+      <label for="schema-id-input">schema id: </label>
+      <select id="table-id-select" v-model="schemaId">
+        <option v-for="option in schemaOptions" :value="option">
+          {{ option }}
+        </option>
+      </select>
+    </div>
+    <div class="m-2">
+      <label for="table-id-select">table id: </label>
```
Comment (davidruvolo51) [inline apps/tailwind-components/pages/DataFetch.other.vue:14]: Is this component used more than once? If so, we need to autogenerate the identifiers

File: apps/tailwind-components/pages/DataFetch.other.vue [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+        </option>
+      </select>
+    </div>
+  </div>
+
+  <div v-if="pending">Loading...</div>
+  <div v-if="error">Error: {{ error }}</div>
+  <div v-if="metadataError">Meta data Error: {{ metadataError }}</div>
+
+  <div v-if="metadata">
```
Comment (davidruvolo51) [inline apps/tailwind-components/pages/DataFetch.other.vue:27]: I think the headings need to be revised. I need to see this in action first.

File: apps/tailwind-components/pages/DisplayList.story.vue [hunk 9]
```diff
@@ -74,3 +66,10 @@ const numberOfDemoItems = ref(3);
     </div>
   </div>
 </template>
+
+<script setup lang="ts">
+const title = ref("My list");
+const columnCount = ref(1);
```
Comment (davidruvolo51) [inline apps/tailwind-components/pages/DisplayList.story.vue:72]: Missing type `number`

File: apps/tailwind-components/pages/DisplayList.story.vue [hunk 10]
```diff
@@ -74,3 +66,10 @@ const numberOfDemoItems = ref(3);
...
+const type: Ref<"standard" | "link"> = ref("standard");
+const numberOfDemoItems = ref(3);
```
Comment (davidruvolo51) [inline apps/tailwind-components/pages/DisplayList.story.vue:74]: Missing type

File: apps/tailwind-components/pages/DisplayList.story.vue [hunk 11]
```diff
@@ -74,3 +66,10 @@ const numberOfDemoItems = ref(3);
     </div>
   </div>
 </template>
+
+<script setup lang="ts">
+const title = ref("My list");
+const columnCount = ref(1);
+const type: Ref<"standard" | "link"> = ref("standard");
```
Comment (davidruvolo51) [inline apps/tailwind-components/pages/DisplayList.story.vue:73]: Should we type variables using the `const var: Ref<...>` approach or using `const var = ref<...>()`?

File: apps/tailwind-components/pages/FilterSearch.story.vue [hunk 12]
```diff
@@ -25,3 +21,8 @@ const inverted = ref(false);
     </div>
   </div>
 </template>
+
+<script setup lang="ts">
+const searchValue = ref("");
```
Comment (davidruvolo51) [inline apps/tailwind-components/pages/FilterSearch.story.vue:26]: Typing is missing

File: apps/tailwind-components/components/content/ContentBlock.vue [hunk 13]
```diff
@@ -0,0 +1,20 @@
+<template>
+  <section
+    class="bg-white py-18 lg:px-12.5 px-5 text-gray-900 xl:rounded-3px last:rounded-b-50px last:mb-18 shadow-primary xl:border-b-0 border-b-[1px] overflow-hidden"
+  >
+    <h2 class="mb-5 uppercase text-heading-4xl font-display" v-if="title">
```
Comment (davidruvolo51) [inline apps/tailwind-components/components/content/ContentBlock.vue:5]: I'm assuming that this component isn't used as a subsection. If see, we would need to make the heading element more flexible.

File: apps/tailwind-components/components/FooterComponent.vue [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+      <a
+        class="text-body-base text-footer-link hover:underline"
+        href="http://molgenis.org"
+        >MOLGENIS.org</a
+      >
+      (<a
+        class="text-body-base text-footer-link hover:underline"
+        href="http://github.com/molgenis/molgenis-emx2"
+      >
+        github </a
```
Comment (davidruvolo51) [inline apps/tailwind-components/components/FooterComponent.vue:14]: There's an extra space in there. In the browser, it returns `( github )`

File: apps/tailwind-components/components/LogoMobile.vue
Comment (davidruvolo51) [inline apps/tailwind-components/components/LogoMobile.vue:1]: I feel like this component could be combined with the Logo component. The logo component could be smatter to show/hide elements based on screen width.

File: apps/tailwind-components/pages/Modal.story.vue [hunk 15]
```diff
@@ -46,3 +36,14 @@ let sectionCount = ref(0);
...
+
+<script setup lang="ts">
+import Modal from "@/components/Modal.vue";
+const modal = ref<InstanceType<typeof Modal>>();
+
+function showModal() {
+  modal.value?.show();
+}
+
+let sectionCount = ref(0);
```
Comment (davidruvolo51) [inline apps/tailwind-components/pages/Modal.story.vue:48]: Missing type `number`

File: apps/tailwind-components/pages/Table.story.vue
Comment (davidruvolo51) [inline apps/tailwind-components/pages/Table.story.vue:1]: This story does not appear in the showcase

File: apps/tailwind-components/components/content/ContentReadMore.vue
[the remark this answers is not in the document]
Comment (davidruvolo51) [reply apps/tailwind-components/components/content/ContentReadMore.vue:1]: The button labels aren't semantic. They should probably say "read more" / "read less". If it is preferred to use the ellipsis, then we should add a label that is visually hidden.

```vue
<button>
    ...
   <span class="sr-only">show more</span>
</button>
```

File: apps/tailwind-components/pages/content/ContentGenericItemList.story.vue
Comment (davidruvolo51) [inline apps/tailwind-components/pages/content/ContentGenericItemList.story.vue:1]: This story isn't visible. Maybe I'm missing something?

File: apps/tailwind-components/pages/content/ContentBlocks.story.vue
Comment (davidruvolo51) [inline apps/tailwind-components/pages/content/ContentBlocks.story.vue:1]: The story isn't shown in the showcase.

**Review by davidruvolo51 — CHANGES_REQUESTED** [review]
Overall, everything looks good. There are some minor things: types missing, lang tags missing, moving CSS computed to tailwind, etc. There are also some larger questions about merging components, rewriting them, etc. 

There are also some larger questions/considerations

- Should header elements be hardcoded in the component? The heading element is determined by the surrounding content.
- How do we want to type variables? `const var: Ref<...>= ref(...)` or `const var = ref<...>(...)`
- Tailwind styles should not be defined using computed values. This sort of defeats the purpose of tailwind and adds unnecessary computations in the browser. Conditionally adding/removing styles should be moved to the `:class` attribute.

File: apps/tailwind-components/components/TableCell.vue [hunk 16]
```diff
@@ -0,0 +1,16 @@
+<!-- eslint-disable vue/multi-word-component-names -->
+<template>
+  <td
+    class="py-2.5 px-2.5 border-b border-gray-200 first:text-blue-500 first:font-bold first:pl-0 last:pr-0 sm:first:pl-2.5 sm:last:pr-2.5"
+  >
+    <div :class="{ 'whitespace-nowrap': allowLineBreak }"><slot></slot></div>
```
[the remark this answers is not in the document]
Comment (MaxPostema) [reply apps/tailwind-components/components/TableCell.vue:6]: I don't think it's against the html5 standard? What would be your alternative suggestion?

File: apps/tailwind-components/components/TableCell.vue [hunk 17]
```diff
@@ -0,0 +1,16 @@
+<!-- eslint-disable vue/multi-word-component-names -->
+<template>
+  <td
```
[the remark this answers is not in the document]
Comment (MaxPostema) [reply apps/tailwind-components/components/TableCell.vue:3]: Good idea, but lets say that is out of scope of a 'moving-components' PR

File: apps/tailwind-components/components/Button.vue [hunk 18]
```diff
@@ -0,0 +1,76 @@
+<template>
+  <button
+    :class="`${colorClasses} ${sizeClasses} ${iconPositionClass} transition-colors`"
```
[the remark this answers is not in the document]
Comment (MaxPostema) [reply apps/tailwind-components/components/Button.vue:3]: I think the whole button will need a good iteration on its self. Lets not do this in this PR.

**Pull request closed without merging**